### PR TITLE
564818 API revision | conditions | key keepers review

### PR DESCRIPTION
**Pull request by eparovyshnaya**
- reimplement `BundleKeykeeper` over the new (1.0) interfaces
 - deprecate obsolete implementation
 - test new `BundleKeykeeper`
 - plug BundleKeyKeeper service to the sealed framework

### CODE REVIEW

File: bundles/org.eclipse.passage.lic.equinox/src/org/eclipse/passage/lic/internal/equinox/i18n/EquinoxMessages.properties [hunk 1]
```diff
@@ -23,6 +23,8 @@ ComponentConfigurationResolver_error_invalid_bundle_context=Unable to extract co
 ComponentConfigurationResolver_error_invalid_component_rt=Unable to extract configuration requirements: invalid ServiceComponentRuntime
 ComponentRequirements_error_no_resource=Unable to extract configuration requirements: invalid {0}
 ComponentRequirements_requirement_for_resource={0} for {1} OSGi-component
+BundleKeyKeeper_failed_to_find_file=Filed to find public key for product [%s] in path [%s] of bundle [%s]
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.equinox/src/org/eclipse/passage/lic/internal/equinox/i18n/EquinoxMessages.properties:26]: F**a**iled

File: bundles/org.eclipse.passage.lic.equinox/src/org/eclipse/passage/lic/internal/equinox/i18n/EquinoxMessages.properties [hunk 2]
```diff
@@ -23,6 +23,8 @@ ComponentConfigurationResolver_error_invalid_bundle_context=Unable to extract co
...
+BundleKeyKeeper_failed_to_open_stream=Filed to open a stream from public key for product [%s] by url [%s]
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.equinox/src/org/eclipse/passage/lic/internal/equinox/i18n/EquinoxMessages.properties:27]: F**a**iled

File: bundles/org.eclipse.passage.lic.equinox/src/org/eclipse/passage/lic/internal/equinox/io/BundleKeyKeeper.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	}
+
+	@Override
+	public LicensedProduct id() {
+		return product.get();
+	}
+
+	@Override
+	public InputStream productPublicKey() throws LicensingException {
+		URL resource = resource(Paths.get("OSGI-INF").resolve(keyFile())); //$NON-NLS-1$
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.equinox/src/org/eclipse/passage/lic/internal/equinox/io/BundleKeyKeeper.java:56]: Should we have `PassageBundlePath` later to encapsulate a way to find the info like this?
We may want to store the licensing information under a subfolder later, like they store bundle properties under `OSGI-INF/l10n`

Comment (eparovyshnaya) [reply bundles/org.eclipse.passage.lic.equinox/src/org/eclipse/passage/lic/internal/equinox/io/BundleKeyKeeper.java:56]: Thought about that, but lots of questions arised. This implementation sticks to the current functionality to keep status quo where it's possible. Let's do it when we actually will need it.

**Pull request merged**